eth/chains/base.py

```diff
@@ -315,13 +315,31 @@ def validate_gaslimit(self, header: BlockHeader) -> None:
     def validate_uncles(self, block: BaseBlock) -> None:
         raise NotImplementedError("Chain classes must implement this method")
 
-    @abstractmethod
+    @classmethod
     def validate_chain(
-            self,
+            cls,
             parent: BlockHeader,
             chain: Tuple[BlockHeader, ...],
             seal_check_random_sample_rate: int = 1) -> None:
-        raise NotImplementedError("Chain classes must implement this method")
+
+        all_indices = list(range(len(chain)))
```

Comment:
Some brief experimentation suggests that the `list` cast isn't needed here.  `range` objects support all of the operations you do below.